client/public/firebase-messaging-sw.js

```diff
@@ -0,0 +1,34 @@
+// This a service worker file for receiving push notifitications.
+// See `Access registration token section` @ https://firebase.google.com/docs/cloud-messaging/js/client#retrieve-the-current-registration-token
+
+// Scripts for firebase and firebase messaging
+importScripts("https://www.gstatic.com/firebasejs/8.2.0/firebase-app.js");
+importScripts("https://www.gstatic.com/firebasejs/8.2.0/firebase-messaging.js");
+
+// Initialize the Firebase app in the service worker by passing the generated config
+const firebaseConfig = {
+  apiKey: "AIzaSyAgUY5d0kc556sShMaMogBmfcLRNZ_ClDU",
+  authDomain: "studydash-fd048.firebaseapp.com",
+  projectId: "studydash-fd048",
+  storageBucket: "studydash-fd048.appspot.com",
+  messagingSenderId: "66459387705",
+  appId: "1:66459387705:web:8e01b1649d8fd481dd8ae8",
+  measurementId: "G-ELZW5L303E",
+};
+
+firebase.initializeApp(firebaseConfig);
+
+// Retrieve firebase messaging
+const messaging = firebase.messaging();
+
+// Handle incoming messages while the app is not in focus (i.e in the background, hidden behind other tabs, or completely closed).
+messaging.onBackgroundMessage(function (payload) {
+  console.log("Received background message ", payload);
+
+  const notificationTitle = payload.notification.title;
+  const notificationOptions = {
+    body: payload.notification.body,
+  };
+
+  self.registration.showNotification(notificationTitle, notificationOptions);
```

Comment:
![image](https://github.com/awctw/study-dash/assets/56493743/ffc7a478-c8c4-4666-82d5-316468394889)

Am receiving the desktop notification twice

Reply:
Encountered the same thing..I have no idea how to fix that yet...will look into it tho

Reply:
fixed